frontend/src/components/LineChart.tsx

```diff
@@ -0,0 +1,50 @@
+import { IconProp } from '@fortawesome/fontawesome-svg-core'
+import dynamic from 'next/dynamic'
+import { useTheme } from 'next-themes'
+import React from 'react'
+import type { ApexChartLabelSeries } from 'types/healthMetrics'
+import AnchorTitle from 'components/AnchorTitle'
+import SecondaryCard from 'components/SecondaryCard'
+// Importing Chart dynamically to avoid SSR issues with ApexCharts
+const Chart = dynamic(() => import('react-apexcharts'), {
+  ssr: false,
+})
+
+const LineChart: React.FC<{
+  title: string
+  series: ApexChartLabelSeries[]
+  labels?: string[]
+  icon?: IconProp
+}> = ({ title, series, labels, icon }) => {
+  const { theme } = useTheme()
+  const color = theme === 'dark' ? '#ececec' : '#1E1E2C'
+
+  return (
+    <SecondaryCard title={<AnchorTitle title={title} />} icon={icon}>
+      <Chart
+        key={theme}
+        options={{
+          chart: {
+            toolbar: {
+              show: false,
+            },
+            foreColor: color,
+          },
+          tooltip: {
+            theme: theme,
+          },
+          xaxis: {
+            categories: labels,
```

Comment:
Let's add `tickAmount` property here to only show maximum of 10 or 15 labels for x axis. 
```suggestion
          xaxis: {
            categories: labels,
            tickAmount: 10,
```
I tested this out and showing all 30 won't work with our chart size (dates are random hardcoded dates):
![Screenshot 2025-06-22 at 6 29 45 PM](https://github.com/user-attachments/assets/4e3f65d4-4ca6-4540-a996-22e84ef0db8e)
![Screenshot 2025-06-22 at 6 29 30 PM](https://github.com/user-attachments/assets/6cede48d-7b83-4556-b2d5-127185d4b29f)